internal/types/constants.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2023 SAP SE or an SAP affiliate company
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+const (
+	// SchemeHTTP indicates a constant for the http scheme
+	SchemeHTTP = "http"
+	// SchemeHTTPS indicates a constant for the https scheme
+	SchemeHTTPS = "https"
+)
```

Comment:
do you see a benefit by defining a new type?

Reply:
Defining a new type can provide some benefits such as type safety and more descriptive code. By creating a new type, you can ensure that only values of that type are used in your code. If we use string, then we are ending up declaring the more duplicate constants. Still with new type, there is chance of duplicate constants, but chances are less.

We already defined duplicate constants here also https://github.com/gardener/etcd-wrapper/pull/3#discussion_r1172542761

I wanted to bring constructing base address in one place instead of helper functions everywhere.

Reply:
Yes i understand the reasons to introduce a new type but i asked this question keeping in view its usage. For now it looks sufficient to let it be a constant. Regarding construction of base address - there are currently 2 URLs that we create - one is for the backup-restore sidecar and other is for the in-process etcd process.

Reply:
Yeah, it's okay to not introducing constants. 
But construction logic of two URLs is same right? If we construct in one place, then we don't need to refer constants other places.

``` golang
func ConstructBaseAddress(hostPort string, tlsEnabled bool) string {
	scheme := SchemeHTTP
	if tlsEnabled {
		scheme = SchemeHTTPS
	}
	return fmt.Sprintf("%s://%s", scheme, hostPort)
}
```

Reply:
sure we can do this.

Reply:
done